tutorials/10_DebuggingREMIND.md

```diff
@@ -0,0 +1,183 @@
+```bash
+Rscript -e "modelstats::loopRuns('.')"
+```
+PIK cluster users can access information on all their runs in all directories by executing `rs2 -c` (or `rs2 -a` for only active runs).
+In case of errors, this tutorial should help you.
+
+Case 1: REMIND did not start
+----------------------------
+
+The first place to look is `log.txt` in the output subfolder for each run. It shows you the log of running [`prepare_and_run.R`](https://github.com/remindmodel/remind/blob/develop/scripts/start/prepare_and_run.R). If it states "Starting REMIND..." somewhere, this shows that the input preparation scripts were successful. If not, this can be caused by the following reasons:
+
+- the model remains locked? start `vi .lock` in the REMIND folder and carefully delete (with `dd`, then save with `:w`) the lines of older runs that may still be present there because you killed these runs.
+- because of a locked folder in the R libraries, outdated packages were loaded: consider [creating a new snapshot](4_RunningREMINDandMAgPIE.md#create-snapshot-of-r-libraries).
+- input files are missing or outdated? delete `input/source_files.log` or set `cfg$force_download <- TRUE` to force the loading of new ones. If this doesn't help, the input generation may be broken.
+- you use an outdated git branch? check `git log` or use main branch with `git checkout main/develop; git pull`.
+- a recent change in `prepare_and_run` may have broken the prepare scripts, check the [file history](https://github.com/remindmodel/remind/commits/develop/scripts/start/prepare_and_run.R) for changes.
+
+Case 2: REMIND did produce an error
+-----------------------------------
+
+In this case, after "REMIND run finished!", you find a short "Model Summary", where the script looks for some files in the output subfolder and gives you its interpretation which is helpful for knowing were to look next:
+```
+  gams_runtime is 10.6 hours.
+  full.gms exists, so the REMIND GAMS code was generated.
+  full.log and full.lst exist, so GAMS did run.
+! abort.gdx exists, containing the latest data at the point GAMS aborted execution.
+  fulldata.gdx exists, so at least one iteration was successful.
+  Number of iterations: 14
+  Modelstat: 4
+! full.log states: *** Status: Execution error(s)
+```
+
+If gams_runtime was short (say: up to 10 minutes), this implies an error right at the beginning of the run.
+If full.gms does not exist, it is an error in the preparation script, not in GAMS.
+
+If full.log exists, this is the next place to look at. Either open it in your favorite text editor or use the console, change directory to the output subfolder and type:
+```bash
+less full.log
+```
+and then type `G` to get to the end of the file. `q` finishes looking at the file.
+Another option is to use the editor `vi` by typing `vi full.log`, in which case `:q` closes the editor. `less` is generally faster when looking at large file, but doesn't offer color coding.
+
+You may find:
+
```

Comment:
After your comments, I added some more information on compilation errors and the different possibilities to restart and reprepare a run.

Reply:
N.B.:
1. You don't get syntax highlighting for .log files in vim either, unless you manually set some file type.
2. The syntax highlighting used for .lst files is GNU Assembler.

Reply:
I deleted the statement about color coding. In the end, the user will have to decide whether using vi, less, notepad or Word, I don't care.